hvsock.go

```diff
@@ -314,3 +316,33 @@ func (conn *HvsockConn) SetReadDeadline(t time.Time) error {
 func (conn *HvsockConn) SetWriteDeadline(t time.Time) error {
 	return conn.sock.SetWriteDeadline(t)
 }
+
+// HvSockRegisterService registers the application defined by the guid and name with the
+// Hyper-V Host's registry.
+//
+// See: https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/user-guide/make-integration-service#register-a-new-application
+func HvSockRegisterService(id guid.GUID, name string) error {
```

Comment:
Can call out if the key already exists this is just a no-op

Reply:
Added that it overwrites the name if the id is registered